### ISSUE TITLE
Alerts over SMTP

### ISSUE THREAD

**Issue by randombugs** [TwiN/gatus#117]
I would really like to send alerts per mail. Is this a possible feature in the future?

Thank you for this nice piece of software!

**Comment by TwinProduction** [comment]
Since both requests are about sending mails, I'll assume that your former suggestion is what you're interested and rename the issue accordingly.

I've definitely been thinking about adding an alerting provider for mails, something like this:
```yaml
alerting:
  mail:
    host: "smtp.gmail.com"
    port: "587"
    from: "from@example.com"
    password: "hunter2"
    to: "to@example.com"
```

**Comment by randombugs** [comment]
This SMTP Mail Alert would be really great! And yes, sending Mail is my main request.

But having the option to execute commands would also allow for 'custom alerts' like sending alerts through other Services then implemented. An Example is Signal Messenger which needs signal-cli for sending messages.

Should I write another request and edit this one for mail only?

**Comment by TwinProduction** [comment]
I'd appreciate if you could edit this one for mail only.

As for the ability to execute custom commands, that's a little tricky, because Gatus' Docker image uses the `scratch` image, which means that adding support for any custom commands would require also modifying the Docker image.

I'm not saying it's off the table, but for now, unless somebody has an actual use for it, I don't think it's worth considering.

**Comment by randombugs** [comment]
Thanks! Seperated the Requests. Shell Command Request is at #118

**Comment by umputun** [comment]
@TwinProduction I have similar email notifications in some of my projects, for example [go-pkgz/auht](https://github.com/go-pkgz/auth/tree/master/provider/sender) and [remark42](https://github.com/umputun/remark42/blob/master/backend/app/notify/email.go). Nothing too tricky here and no external dependencies involved, but maybe it will save you some time.

Maybe @paskal (he is the contributor handled email part) can submit a PR for this, but either way, the implementation should be straightforward

**Comment by KirschbaumP** [comment]
+1

**Comment by zeylos** [comment]
@TwinProduction I'll also need that, do you want me to work on this one ?

**Comment by TwinProduction** [comment]
@zeylos If you're still up for the challenge, go for it!
(sorry for the delay, just noticed this message now 😅)

**Comment by TwiN** [comment]
Alright fellas, this has been a requested feature for long enough, and at long last, here it is.

**Comment by TwiN** [comment]
Released in [v3.4.0](https://github.com/TwiN/gatus/releases/tag/v3.4.0)
